MediaToolkit src/MediaToolkit/Util/RegexEngine.cs

```diff
@@ -53,22 +53,59 @@ internal static bool IsProgressData(string data, out ConvertProgressEventArgs pr
             Match matchTime = Index[Find.ConvertProgressTime].Match(data);
             Match matchBitrate = Index[Find.ConvertProgressBitrate].Match(data);
 
-            if (!matchFrame.Success || !matchFps.Success || !matchSize.Success || !matchTime.Success ||
-                !matchBitrate.Success) return false;
+            if (!matchSize.Success || !matchTime.Success || !matchBitrate.Success)
+                return false;
 
             TimeSpan processedDuration;
             TimeSpan.TryParse(matchTime.Groups[1].Value, out processedDuration);
 
-            long frame = Convert.ToInt64(matchFrame.Groups[1].Value, CultureInfo.InvariantCulture);
-            double fps = Convert.ToDouble(matchFps.Groups[1].Value, CultureInfo.InvariantCulture);
-            int sizeKb = Convert.ToInt32(matchSize.Groups[1].Value, CultureInfo.InvariantCulture);
-            double bitrate = Convert.ToDouble(matchBitrate.Groups[1].Value, CultureInfo.InvariantCulture);
+            long? frame = GetLongValue(matchFrame);
+            double? fps = GetDoubleValue(matchFps);
+            int? sizeKb = GetIntValue(matchSize);
+            double? bitrate = GetDoubleValue(matchBitrate);
 
             progressEventArgs = new ConvertProgressEventArgs(processedDuration, TimeSpan.Zero, frame, fps, sizeKb, bitrate);
 
             return true;
         }
 
+        private static long? GetLongValue(Match match)
+        {
+            try
```

Comment:
Try statements are a bit heavy.  I would recommend doing something like this instead:

long parsed;
if (long.TryParse(match.Groups[1].value, out parsed)) {
return parsed;
}
else {
return null;
}